citations-tool/tool/src/java/org/sakaiproject/citation/tool/CitationHelperAction.java

```diff
@@ -2472,6 +2486,45 @@ public String buildNewResourcePanelContext(VelocityPortlet portlet, Context cont
 			contentProperties = new HashMap<String,Object>();
 		}
 		context.put("contentProperties", contentProperties);
+
+
+		BufferedReader streamReader = null;
+		try
+		{
+			ContentResource cr = contentService.getResource(MANAGING_LIBRARIES_JSON);
+			streamReader = new BufferedReader(new InputStreamReader(cr.streamContent(), "UTF-8"));
+			StringBuilder responseStrBuilder = new StringBuilder();
+
+			String inputStr;
+			while ((inputStr = streamReader.readLine()) != null){
+				responseStrBuilder.append(inputStr);
+			}
+			JSONObject jsonObject = JSONObject.fromObject(responseStrBuilder.toString());
```

Comment:
If the JSON is badly formed this will throw an exception and stop the rest of the method running, is it worth catching this and allowing the method to carry on running?